### PR TITLE
delete calendars when users are deleted

### LINKED ISSUE #25912: Calendars are not deleted when an user is deleted
### Steps to reproduce
1. Create an user. The default calendar is created for him/her
2. Delete the user

### Expected behaviour
Calendar should be deleted inside database

### Actual behaviour
Calendar is not deleted in database (`oc_calendars` table)

### Server configuration
**Database:** SQLite

**PHP version:** 7.0

**ownCloud version:** (see ownCloud admin page) git master

### PR DESCRIPTION
**Pull request by tcitworld**
Fixes #25912

No tests changed, maybe time to add some ?

### CODE REVIEW

File: apps/dav/lib/HookManager.php [hunk 1]
```diff
@@ -98,6 +101,7 @@ public function preDeleteUser($params) {
 		$this->usersToDelete[$params['uid']] = $user;
 
 		$this->calendarsToDelete = $this->calDav->getCalendarsForUser('principals/users/' . $user->getUID());
+		$this->addressBooksToDelete = $this->cardDav->getAddressBooksForUser('principals/users/' . $user->getUID());
```
Comment (DeepDiver1975) [inline apps/dav/lib/HookManager.php]: No need to cache the books here. Just retrieve the books in the post hook.

**Comment by DeepDiver1975** [comment]
> No tests changed, maybe time to add some ?

absolutly - give it a shot ... let me know if you need support ....

**Comment by PVince81** [comment]
Must calendar shares for these also be deleted somehow ? Or is that done by a cron job that cleans up orphaned shares ?

**Comment by DeepDiver1975** [comment]
> Must calendar shares for these also be deleted somehow ?

nice catch .... we should add this as well ...

**Comment by tcitworld** [comment]
Added, but the whole doesn't work. Also, I'm waiting for https://github.com/owncloud/core/pull/26200 to be merged.

**Comment by tcitworld** [comment]
```
Exception: {
"Exception":"Sabre\\DAV\\Exception",
"Message":"Principal not found",
"Code":0,
"Trace":"
#0 \/home\/tcit\/dev\/core\/apps\/dav\/lib\/CalDAV\/CalDavBackend.php(211): OCA\\DAV\\Connector\\Sabre\\Principal->getGroupMembership('principals\/user...', true)\n
#1 \/home\/tcit\/dev\/core\/apps\/dav\/lib\/HookManager.php(102): OCA\\DAV\\CalDAV\\CalDavBackend->getCalendarsForUser('principals\/user...')\n
#2 \/home\/tcit\/dev\/core\/lib\/private\/legacy\/hook.php(105): OCA\\DAV\\HookManager->postDeleteUser(Array)\n
#3 \/home\/tcit\/dev\/core\/lib\/private\/Server.php(254): OC_Hook::emit('OC_User', 'post_deleteUser', Array)\n
#4 [internal function]: OC\\Server->OC\\{closure}(Object(OC\\User\\User))\n
#5 \/home\/tcit\/dev\/core\/lib\/private\/Hooks\/EmitterTrait.php(98): call_user_func_array(Object(Closure), Array)\n
#6 \/home\/tcit\/dev\/core\/lib\/private\/Hooks\/PublicEmitter.php(32): OC\\Hooks\\BasicEmitter->emit('\\\\OC\\\\User', 'postDelete', Array)\n
#7 \/home\/tcit\/dev\/core\/lib\/private\/User\/User.php(220): OC\\Hooks\\PublicEmitter->emit('\\\\OC\\\\User', 'postDelete', Array)\n
#8 \/home\/tcit\/dev\/core\/settings\/Controller\/UsersController.php(467): OC\\User\\User->delete()\n
#9 [internal function]: OC\\Settings\\Controller\\UsersController->destroy('tcit1')\n
#10 \/home\/tcit\/dev\/core\/lib\/private\/AppFramework\/Http\/Dispatcher.php(159): call_user_func_array(Array, Array)\n
#11 \/home\/tcit\/dev\/core\/lib\/private\/AppFramework\/Http\/Dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OC\\Settings\\Controller\\UsersController), 'destroy')\n
#12 \/home\/tcit\/dev\/core\/lib\/private\/AppFramework\/App.php(99): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OC\\Settings\\Controller\\UsersController), 'destroy')\n
#13 \/home\/tcit\/dev\/core\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php(46): OC\\AppFramework\\App::main('UsersController', 'destroy', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n
#14 [internal function]: OC\\AppFramework\\Routing\\RouteActionHandler->__invoke(Array)\n
#15 \/home\/tcit\/dev\/core\/lib\/private\/Route\/Router.php(280): call_user_func(Object(OC\\AppFramework\\Routing\\RouteActionHandler), Array)\n
#16 \/home\/tcit\/dev\/core\/lib\/base.php(896): OC\\Route\\Router->match('\/settings\/users...')\n
#17 \/home\/tcit\/dev\/core\/index.php(39): OC::handleRequest()\n
#18 {main}","File":"\/home\/tcit\/dev\/core\/apps\/dav\/lib\/Connector\/Sabre\/Principal.php","Line":145}
```

**Comment by tcitworld** [comment]
Getting list of calendars to delete is to be done in the `preDeleteUser`, otherwise we get `Principal not found` since the user doesn't exist anymore.
Also, we don't want to delete calendars which are shared to the user, so we don't use `getCalendarsForUser()` anymore. There's certainly a better solution that the function I brought, though.

**Comment by DeepDiver1975** [comment]
tested - works :+1:

**Comment by koenlek** [comment]
We have orphaned calenders (or more importantly: calendar shares) which I would like to delete. It's fine with me to delete them manually from the database.

Any instructions how I should do that?

**Pull request merged**